src/PoolManager.sol

```diff
@@ -135,35 +130,33 @@ contract PoolManager is IPoolManager, Fees, NoDelegateCall, ERC6909Claims {
 
     /// @inheritdoc IPoolManager
     function lock(address lockTarget, bytes calldata data) external payable override returns (bytes memory result) {
-        Lockers.push(lockTarget, msg.sender);
+        // Get the lock caller because thats an EOA and is not user-controlable
+        if (Locker.isLocked()) revert AlreadyLocked();
```

Comment:
Will disallowing multiple locks break composability?? i.e. router has to open a lock, as well as any other integrating contract that's not a hook. What if they get called after another non-hook contract has already locked it, they now have 2 ways they'd need to integrate?

Reply:
building on this ---^ the word `lock` seems like a misnomer now, in fact it's like the opposite, now it's open for business (right?? 😄 ) Wondering if we need to change our mental model if this is the direction we're going.....

Reply:
if the gas savings are for hooks specifically, wonder if we can allow multiple locks, but free up hooks from needing to use them? (really just spitballing/food-for-thought, but just making sure we've thought through this new mental model of locks)

Reply:
>Will disallowing multiple locks break composability??

Yeah I think that integrating contracts who want to be able to be a sub-lock too will need to write something more like this to just jump straight to their execution logic if someone else has already called lock at an upper level
```
if (!poolManager.isLocked) {
   poolManager.lock(lockData);
} else {
   lockAcquired(lockData)
}
```

Thoughts? Do you think thats too janky?

Reply:
And yes I agree the word `lock` feels completely wrong haha

Reply:
yah, it's annoying boilerplate but ultimately saves gas, and I'm having trouble finding a good reason to keep lockers nested or stack/recursed like it was way back.

 Although the lock vernacular doesn't seem to make much sense anymore..... but naming is so hAaRrdddDD......

```
if (!poolManager.ledgerOpened?) {
   poolManager.openLedger(ledgerData);
} else {
   ledgerCallback(ledgerData)
}
```

??

```
if (!poolManager.settlementInitiated) {
   poolManager.openSettlement(settlementData);
} else {
   settlementCallback(settlementData)
}```